<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TimeSyncPanel</class>
 <widget class="QDialog" name="TimeSyncPanel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>720</width>
    <height>580</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Time Synchronization</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QDialog { background-color: #2b2b2b; color: #ddd; }
QGroupBox { border: 1px solid #555; border-radius: 4px; margin-top: 8px; padding-top: 12px; color: #ccc; }
QGroupBox::title { subcontrol-origin: margin; left: 10px; padding: 0 4px; }
QPushButton { background-color: #444; color: #ddd; border: 1px solid #666; border-radius: 3px; padding: 4px 12px; }
QPushButton:hover { background-color: #555; }
QPushButton:disabled { background-color: #333; color: #666; border: 1px solid #444; }
QCheckBox { color: #ddd; }
QLineEdit { background-color: #333; color: #ddd; border: 1px solid #555; border-radius: 2px; padding: 2px 4px; }
QLabel { color: #ddd; }</string>
  </property>
  <layout class="QHBoxLayout" name="mainHLayout">
   <property name="spacing">
    <number>10</number>
   </property>
   <property name="leftMargin">
    <number>10</number>
   </property>
   <property name="topMargin">
    <number>10</number>
   </property>
   <property name="rightMargin">
    <number>10</number>
   </property>
   <property name="bottomMargin">
    <number>10</number>
   </property>
   <!-- Left column: existing content -->
   <item>
    <layout class="QVBoxLayout" name="leftColumn">
     <property name="spacing">
      <number>6</number>
     </property>
     <item>
      <widget class="QLabel" name="lblUtcClock">
       <property name="text">
        <string>00:00:00.000 UTC</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="styleSheet">
        <string notr="true">font-family: Consolas, 'Courier New', monospace; font-size: 24pt; font-weight: bold; color: #00ff00; background-color: #1a1a1a; padding: 8px; border-radius: 4px;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="lblUtcDate">
       <property name="text">
        <string>2026-01-01</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="styleSheet">
        <string notr="true">font-size: 14pt; color: #aaa;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="lblSystemClock">
       <property name="text">
        <string>System: 00:00:00.000</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="styleSheet">
        <string notr="true">font-size: 10pt; color: #888;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupNtp">
       <property name="title">
        <string>NTP Status</string>
       </property>
       <layout class="QFormLayout" name="formLayoutNtp">
        <item row="0" column="0">
         <widget class="QLabel" name="labelStatus">
          <property name="text">
           <string>Status:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="lblNtpStatus">
          <property name="text">
           <string>--</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelOffset">
          <property name="text">
           <string>Offset:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="lblNtpOffset">
          <property name="text">
           <string>--</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="labelLastSync">
          <property name="text">
           <string>Last sync:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="lblLastSync">
          <property name="text">
           <string>--</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupControls">
       <property name="title">
        <string>Controls</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayoutControls">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayoutControls">
          <item>
           <widget class="QCheckBox" name="cbNtpEnabled">
            <property name="text">
             <string>NTP Enabled</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="btnSyncNow">
            <property name="text">
             <string>Sync Now</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayoutServer">
          <item>
           <widget class="QLabel" name="labelServer">
            <property name="text">
             <string>Server:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="editCustomServer">
            <property name="placeholderText">
             <string>(auto - pool.ntp.org)</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupHistory">
       <property name="title">
        <string>Offset History</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayoutHistory">
        <item>
         <widget class="QListWidget" name="listHistory">
          <property name="minimumHeight">
           <number>80</number>
          </property>
          <property name="styleSheet">
           <string notr="true">QListWidget { background-color: #1a1a1a; color: #ddd; border: 1px solid #444; font-family: Consolas, 'Courier New', monospace; font-size: 9pt; }</string>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::NoSelection</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnClearHistory">
       <property name="text">
        <string>Clear History</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <!-- Right column: ChronoGPS panel -->
   <item>
    <layout class="QVBoxLayout" name="rightColumn">
     <property name="spacing">
      <number>8</number>
     </property>
     <item>
      <widget class="QGroupBox" name="groupChronoGPS">
       <property name="title">
        <string>ChronoGPS v2.4.6</string>
       </property>
       <property name="styleSheet">
        <string notr="true">QGroupBox#groupChronoGPS { border: 1px solid #4a7a9b; }</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayoutChrono">
        <property name="spacing">
         <number>8</number>
        </property>
        <item>
         <widget class="QLabel" name="lblChronoIcon">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="minimumSize">
           <size>
            <width>64</width>
            <height>64</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblChronoTitle">
          <property name="text">
           <string>ChronoGPS v2.4.6</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 14pt; font-weight: bold; color: #fff;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblChronoDesc">
          <property name="text">
           <string>GPS/NTP Time Synchronization</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #aaa;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblChronoAuthor">
          <property name="text">
           <string>by Yoshiharu Tsukuura, JP1LRT</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #888; font-style: italic;</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="chronoSpacer1">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>10</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="lblChronoStatus">
          <property name="text">
           <string>Stopped</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 12pt; font-weight: bold; color: #ff4444;</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="chronoSpacer2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>10</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="btnLaunchChrono">
          <property name="text">
           <string>Launch ChronoGPS</string>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton { padding: 8px 16px; font-size: 11pt; }</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnStopChrono">
          <property name="text">
           <string>Stop ChronoGPS</string>
          </property>
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">QPushButton { padding: 8px 16px; font-size: 11pt; }</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="chronoSpacer3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>6</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="cbChronoAutoLaunch">
          <property name="text">
           <string>Auto-launch with Decodium</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupChronoInfo">
       <property name="title">
        <string>Info</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayoutChronoInfo">
        <item>
         <widget class="QLabel" name="lblChronoInfo">
          <property name="text">
           <string>ChronoGPS provides precise GPS/NTP time synchronization for digital modes (FT2/FT4/FT8).

Supports GPS, GLONASS, BeiDou, Galileo, SBAS, QZSS receivers + NTP.</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #aaa; font-size: 9pt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblChronoLink">
          <property name="text">
           <string>&lt;a href=&quot;https://github.com/jp1lrt/ChronoGPS&quot; style=&quot;color: #5dade2;&quot;&gt;github.com/jp1lrt/ChronoGPS&lt;/a&gt;</string>
          </property>
          <property name="openExternalLinks">
           <bool>true</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 9pt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="rightColumnSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
